liblangutil/EVMVersion.cpp

```diff
@@ -49,6 +49,11 @@ bool EVMVersion::hasOpcode(Instruction _opcode) const
 		return hasSelfBalance();
 	case Instruction::BASEFEE:
 		return hasBaseFee();
+	case Instruction::PREVRANDAO:
+	{
+		std::string name = instructionInfo(Instruction::PREVRANDAO, *this).name;
+		return (name == "PREVRANDAO" && supportsPrevRandao()) || (name == "DIFFICULTY" && !supportsPrevRandao());
```

Comment:
This is a convoluted way of saying ``true``, isn't it :-)?

Reply:
haha it sounds like it, but actually no, since it will be `false` if either holds:

- `difficulty` is used with `evm >= paris`
- `prevrandao` is used with `evm < paris`

Those are the scenarios where the warning is triggered.

Reply:
Is that really true :-)? The argument of this function is always ``Instruction:::PREVRANDAO``, since we don't have ``Instruction::DIFFICULTY`` - given that you in general can't distinguish between the two here.
Now ``instructionInfo(Instruction::PREVRANDO, *this).name`` will be exactly ``"PREVRANDAO"``, if ``supportsPrevRandao()`` is true - and it will be exactly ``"DIFFICULTY"``, if it's false.

So I'm pretty sure this will actually always return ``true`` here :-).
Which is fine, the instruction is always there after all - you can remove the special case for ``Instruction::PREVRANDAO`` entirely here.

Reply:
Oh! Yes, I see it now, you are right haha
I will remove it.